src/sentry/taskworker/client.py

```diff
@@ -142,6 +149,23 @@ def _get_cur_stub(self) -> tuple[str, ConsumerServiceStub]:
         if self._num_tasks_before_rebalance == 0:
             self._cur_host = random.choice(self._hosts)
             self._num_tasks_before_rebalance = self._max_tasks_before_rebalance
+            self._num_consecutive_unavailable_errors = 0
+            metrics.incr(
+                "taskworker.client.loadbalancer.rebalance",
+                tags={"reason": "max_tasks_reached"},
+            )
+        elif (
+            self._num_consecutive_unavailable_errors == self._max_consecutive_unavailable_errors
```

Comment:
```suggestion
            self._num_consecutive_unavailable_errors >= self._max_consecutive_unavailable_errors
```

Just in case we end up with more errors than the threshold.